pep-9999.rst

```diff
@@ -0,0 +1,294 @@
+PEP: 9999
+Title: Inlined comprehensions
+Author: Carl Meyer <carl@oddbird.net>
+Sponsor: Guido van Rossum <guido@python.org>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 24-Feb-2023
+Python-Version: 3.12
+Post-History:
+
+
+Abstract
+========
+
+Comprehensions are currently compiled as nested functions, which provides
+isolation of the comprehension's iteration variable, but is inefficient at
+runtime. This PEP proposes to inline list, dictionary, and set comprehensions
+into the function where they are defined, and provide the expected isolation by
+pushing/popping clashing locals on the stack. This change makes comprehensions
+much faster: up to 2x faster for a microbenchmark of a comprehension alone,
+translating to 11% speedup for one sample benchmark derived from real-world code
+that makes heavy use of comprehensions in the context of doing actual work.
+
+
+Motivation
+==========
```

Comment:
Not a huge deal, but FWIW, per PEP 1, PEP 12 and general convention, this section is IMO more Rationale than Motivation, as it mostly discusses the reasoning behind the proposed solution and the design decisions it makes rather than explains the motivation problem (performance, which is fairly self-evident).

```suggestion
Rationale
=========
```